opentelemetry-api/src/opentelemetry/context/propagation/binaryformat.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import typing
+
+from opentelemetry.trace import SpanContext
+
+
+class BinaryFormat(abc.ABC):
```

Comment:
In order to not duplicate the same class for correlation-context (distributedcontext) I would suggest to make this a template and have BinaryFormat<SpanContext> and BinaryFormat<DistributedContext>

Reply:
Thanks! Mypy typing does support generics, but I'm planning of filing another PR that aligns closer to the tickets I've mentioned above. This will include creating a unified composed object for SpanContext and DistributedContext and using the composed object as the context for propagators, thereby eliminating the need for the generic.

Reply:
From an offline conversation with @bogdandrutu: a bigger problem is that referencing `SpanContext` here would make `context` depend on `trace`.

This creates a circular dependency, and even if we remove the propagators from the tracer `trace` will depend on `context` in other places. If we think of the `context` package as representing the [base context propagation layer](https://docs.google.com/document/d/1UxrEYOaQlF_E4gtiPoFmcZ4YKKe1GxohvCvQDuwvD1I/edit#heading=h.i5vdaou65br) it's surprising to see it depend on a layer higher in the stack.

This isn't a blocking comment for this PR, just something to keep in mind when you start composing context objects.